### DISCUSSION TITLE
Channels mass import

### DISCUSSION THREAD

**Discussion by radeknoone** [hmatuschek/qdmr#214] [Q&A]
Hi, I'm using version 10 and trying to figure-out howto import a lot of channels I have in csv format. In the qdmr is no option to "import from csv" (neither other mass import). So, I was used to insert channels directly into the codeplug file in old format which was pretty easy. Now the standard format is yaml and it is much harder to prepare channels and modify them to match the yaml format.

My worklfow is like:
I have channels in csv/xls/txt (as I'm moviing around the world and often adding/removing/modifying channels and zones).
I have to transform it to the yaml file format and upload to radio.

Is there any possibility howt to do it effectively - eeg import from csv or "csv2yaml converter"?

Thank you!

**Comment by hmatuschek** [comment]
For now, there is no means to mass-import channels from a CSV file. The major problem is, that there is no standard format which column means what. To this end an import gets difficult. The easiest was is likely to use the old format. QDMR can still import this table based format but doe not export it. 

Alternatively, you may try to write a small script in Python that generates the corresponding YAML file. The YAML format is somewhat well described in https://dm3mat.darc.de/qdmr/manual/ch03.html.
